pkg/cmd/repo/autolink/list/http.go

```diff
@@ -0,0 +1,56 @@
+package list
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghinstance"
+	"github.com/cli/cli/v2/internal/ghrepo"
+)
+
+type AutolinkGetter struct {
+	HttpClient *http.Client
+}
+
+func NewAutolinkGetter(httpClient *http.Client) *AutolinkGetter {
+	return &AutolinkGetter{
+		HttpClient: httpClient,
+	}
+}
+
+func (a *AutolinkGetter) Get(repo ghrepo.Interface) ([]autolink, error) {
+	path := fmt.Sprintf("repos/%s/%s/autolinks", repo.RepoOwner(), repo.RepoName())
+	url := ghinstance.RESTPrefix(repo.RepoHost()) + path
```

Comment:
**Question**

Should we be using `ghrepo.GenerateRepoURL` here, as is done in the command itself?

Reply:
I could definitely be confused about what's going on here, but I don't _think_ we can? `ghinstance.RESTPrefix` supplies the `api` subdomain (or path component in the case of an enterprise instance). This allows us to build the correct URL for the REST endpoint we want to hit here. In the command, we use `ghrepo.GenerateRepoURL` to generate the URL for the web interface that displays the autolinks. So it's the difference between:

* Current implementation: `https://api.github.com/repos/OWNER/REPO/autolinks`
* `ghrepo.GenerateRepoURL(repo, "autolinks")`: `https://github.com/OWNER/REPO/autolinks`

Maybe there's some other way to use `GenerateRepoURL` to get the REST endpoint though?